### PR TITLE
Bg 79327 provenance sdk addition

### PR DESCRIPTION
**Pull request by DinshawKothari**

### CODE REVIEW

File: modules/sdk-coin-hash/package.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
+  "name": "@bitgo/sdk-coin-hash",
+  "version": "1.0.0",
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-hash/package.json:3]: Will this be ready for publish or should "access": "public" be private until then?

File: modules/sdk-coin-hash/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    {
+      "path": "../sdk-api"
+    },
+    {
+      "path": "../sdk-core"
+    },
+    {
+      "path": "../sdk-test"
+    },
+    {
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-hash/tsconfig.json:22]: Missing statics

**Review by mmcshinsky-bitgo — COMMENTED** [review]
Missing definitions from `statics`?

**Comment by DinshawKothari** [comment]
> Missing definitions from `statics`?

This is already completed in another PR.

File: modules/sdk-coin-hash/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+const sdk = new BitGoAPI();
+
+sdk.register('hash', Hash.createInstance);
+```
+
+## Development
+
+Most of the coin implementations are derived from `@bitgo/sdk-core`, `@bitgo/statics`, and coin specific packages. These implementations are used to interact with the BitGo API and BitGo platform services.
+
+You will notice that the basic version of common class extensions have been provided to you and must be resolved before the package build will succeed. Upon initiation of a given SDK coin, you will need to verify that your coin has been included in the root `tsconfig.packages.json` and that the linting, formatting, and testing succeeds when run both within the coin and from the root of BitGoJS.
```
Comment (OttoAllmendinger) [inline modules/sdk-coin-hash/README.md:30]: please add a link to the website of this project and the developer resources

**Review by OttoAllmendinger — DISMISSED** [review]

File: modules/sdk-coin-hash/test/resources/hash.ts [hunk 4]
```diff
@@ -0,0 +1,226 @@
+export const TEST_ACCOUNT = {
```
Comment (OttoAllmendinger) [inline modules/sdk-coin-hash/test/resources/hash.ts:1]: where did you get these from?

File: modules/sdk-coin-hash/src/lib/constants.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export const validDenoms = ['nhash', 'uhash', 'mhash', 'hash'];
+export const accountAddressRegex = /^(tp)1(['qpzry9x8gf2tvdw0s3jn54khce6mua7l']+)$/;
+export const validatorAddressRegex = /^(tpvaloper)1(['qpzry9x8gf2tvdw0s3jn54khce6mua7l']+)$/;
```
Comment (OttoAllmendinger) [inline modules/sdk-coin-hash/src/lib/constants.ts:3]: source?

Comment (mukeshsp) [reply modules/sdk-coin-hash/src/lib/constants.ts:3]: @DinshawKothari 
Prefix of validator address is different for test and prod.
Please have a look the validator address
Test: https://explorer.test.provenance.io/validators
Prod: https://explorer.provenance.io/validators

File: modules/sdk-coin-hash/tsconfig.json (hunk 2 above)
Comment (DinshawKothari) [reply modules/sdk-coin-hash/tsconfig.json:25]: Added.

File: modules/sdk-coin-hash/test/resources/hash.ts (hunk 4 above)
Comment (OttoAllmendinger) [reply modules/sdk-coin-hash/test/resources/hash.ts:1]: please add comment

**Review by Vijay-Jagannathan — APPROVED** [review]

File: modules/sdk-coin-hash/README.md (hunk 3 above)
Comment (DinshawKothari) [reply modules/sdk-coin-hash/README.md:30]: added.

**Pull request merged**